crates/wac-types/src/package.rs

```diff
@@ -154,23 +164,34 @@ pub struct Package {
 }
 
 impl Package {
+    /// Gets the package key for the package.
+    pub fn key(&self) -> BorrowedPackageKey {
+        BorrowedPackageKey::new(self)
+    }
+
     /// Creates a new package from the given file path.
+    ///
+    /// The package will populate its types into the provided type collection.
     pub fn from_file(
         name: &str,
         version: Option<&Version>,
         path: impl AsRef<Path>,
+        types: &mut Types,
     ) -> Result<Self> {
         let path = path.as_ref();
         let bytes = std::fs::read(path)
             .with_context(|| format!("failed to read `{path}`", path = path.display()))?;
-        Self::from_bytes(name, version, Arc::new(bytes))
+        Self::from_bytes(name, version, bytes, types)
     }
 
     /// Creates a new package from the given bytes.
+    ///
+    /// The package will populate its types into the provided type collection.
     pub fn from_bytes(
         name: &str,
         version: Option<&Version>,
-        bytes: impl Into<Arc<Vec<u8>>>,
+        bytes: impl Into<Vec<u8>>,
+        types: &mut Types,
```

Comment:
Even though it's not necessary... I wonder if the `wac-graph` crate should define a new type for `Package` that wraps this one and ensures the `Types` provided comes from the graph...

Reply:
For my understanding, could you explain a bit more what this might look like?

Reply:
```rust
struct Package(wac_types::Package);

impl Package {
   pub fn from_bytes(
        name: &str,
        version: Option<&Version>,
        bytes: impl Into<Vec<u8>>,
       graph: &mut CompositionGraph
 ) -> Result<Self> {
  Self(wac_types::Package::from_bytes(name, version, bytes, graph.types_mut())
}
```

This is probably overkill - I'm just worried about people being confused about how to supply the types collection.